fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/ScopedDirectoryLayer.java

```diff
@@ -149,6 +149,22 @@ private CompletableFuture<Optional<ResolverResult>> readInternal(@Nonnull FDBRec
                 CompletableFuture.completedFuture(Optional.empty()));
     }
 
+    @Override
+    protected CompletableFuture<Optional<String>> readReverse(@Nonnull FDBRecordContext context, Long value) {
+        FDBReverseDirectoryCache reverseCache = database.getReverseDirectoryCache();
+        return reverseCache.get(context, wrap(value));
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param timer a timer to use to instrument this operation
+     * @param value the value returned by this resolver
+     * @return a future returning an optional set to the key associated with the given value or {@link Optional#empty()}
+     *     if there is no such value in the map
+     * @deprecated implementors should switch to using {@link #readReverse(FDBRecordContext, Long)} instead
+     */
+    @Deprecated
```

Comment:
Should this implementation be removed?

Reply:
I don't think we can yet... This is maybe a bit messy, but the easiest time to remove it, I think, would be when we remove the method from `LocatableResolver.java`. At present, callers can still end up using this method if they use the (now deprecated) method on `LocatableResolver`.

At present, `LocatableResolver` is written so that the new method (with a context) calls the old method (with a timer) to avoid breaking external extenders. If `LocatableResolver` were changed to instead make the old method (with a timer) call the new method (with a context) and made the new method `abstract`, then we could  in fact get rid of these implementations, though I think they'd be slightly less efficient than they are today (because they'd need to make some contexts just to throw them away without using them).